spec/src/main/asciidoc/cdi.asciidoc

```diff
@@ -17,6 +17,75 @@
 [[concurrencycdi]]
 == CDI Injection
 
-If a CDI implementation is available, instances of `ManagedExecutor` and `ThreadContext` can be injected as CDI beans.
+If a CDI implementation is available, instances of `ManagedExecutor` and `ThreadContext` can be injected as CDI beans. 
 
-TODO write the documentation after discussion on Config annotations is resolved
\ No newline at end of file
+=== Injecting ManagedExecutor
+
+The application can annotate injection points of `ManagedExecutor` with both, either or neither of,
+
+- `@NamedInstance` - This annotation is a standard CDI qualifier. It has a single value, which is the unique name for a `ManagedExecutor` instance. All injection points within an application that specify the same name inject the same instance.
+
+- `@ManagedExecutorConfig` - This annotation supplies configuration for an instance of `ManagedExecutor` that is built and managed by the container. The same instance is shared within an application across injection points that have the same unique name. The unique name is determined by the value `@NamedInstance` annotation if present. Otherwise, it is generated from the fully qualified class name and field name of the injection point, with the `.` character as the delimiter.
+
+When an injection point for `ManagedExecutor` is annotated with only the `@NamedInstance` qualifier, the container injects an existing instance of `ManagedExecutor` that is shared across all injection points in the application that have the same name.
+
+When an injection point for `ManagedExecutor` is annotated with only the `@ManagedExecutorConfig` annotation, the container builds a single instance of `ManagedExecutor` matching the configuration attributes of the `@ManagedExecutorConfig` annotation. The container injects this same instance into the annotated injection point, managing its life cycle, and shutting it down when the application stops.
+
+When an injection point for `ManagedExecutor` is annotated with both `@ManagedExecutorConfig` and `@NamedInstance`, the container builds a single instance of `ManagedExecutor` matching the configuration attributes of the `@ManagedExecutorConfig` annotation. The container injects this same instance into the annotated injection point, in addition to all other injection points that are annotated with `@NamedInstance` that specify the same name. The container manages the life cycle of the `ManagedExecutor` instance, shutting it down when the application stops.
+
+When an injection point for `ManagedExecutor` is annotated with neither of the above, the container injects a default instance of `ManagedExecutor` which is shared within the application across all similar injection points (annotated with neither of the above). The configuration of this instance matches the default attribute values that are used by `@ManagedExecutorConfig()`.
```

Comment:
`...which is shared within the application across all similar injection points...`
This goes against the default sharing you and @aguibert argued against. It is also contradictory to only sharing instances with same name on them and gives you no way to request several different default executors in your application.

Reply:
I'm fine with removing the language about sharing here.  I was thinking it might be more efficient given that the default configuration for ManagedExecutor has unlimited maxAsync/maxQueued (unconstrained) and so which instance is used wouldn't make much difference in that case.  However, if MP Config is used to override, then there could be concurrency constraints configured and we would need separate instances after all.